### DISCUSSION TITLE
Can pyo3 build with x86_64 conda env on macos arm64?

### DISCUSSION THREAD

**Discussion by pymongo** [PyO3/pyo3#2566] [Questions] [answered]
pyo3 can compile with arm64 conda env on my mac, but when I share my binary to other mac M1 with x86_64 conda env got a error:

> libpython3.9.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e

So I switch my conda env to x64(`CONDA_SUBDIR=osx-64 conda create -n intel python=3.9`) and recompile but get these error

```
  = note: ld: warning: ignoring file /Users/chenyufeng/opt/miniconda3/envs/intel/lib/libpython3.9.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
          Undefined symbols for architecture arm64:
            "__Py_TrueStruct", referenced from:
                pyo3_ffi::boolobject::Py_True::hf994be6c87cc506f in libidp_kernel-f0e92cd67d4730ad.rlib(idp_kernel-f0e92cd67d4730ad.4l6pzkjdgdagba5r.rcgu.o)
```

**Comment by davidhewitt** [comment]
Are you using setuptools-rust or maturin? Sounds like something those projects should be able to handle.

**Comment by pymongo** [comment, marked as answer]
I vendor arm64 libpython3.9.dylib with my binary, and force conda x64 env use my vendor dylib, my binary works well on arm64 with conda x64 env
